packages/openactive-broker-microservice/src/sample-requests.js

```diff
@@ -0,0 +1,237 @@
+    "addressCountry": "GB"
+  }
+};
+
+const customer = {
+  "@type": "Person",
+  "email": "geoffcapes@example.com",
+  "telephone": "020 811 8055",
+  "givenName": "Geoff",
+  "familyName": "Capes"
+};
+
+function getSampleRequestConstraints(opportunity) {
+  return merge(sampleRequestConstraints, opportunity);
+}
+
+function checkpointOne(opportunity, seller, offer) {
+  return {
+    "@context": "https://openactive.io/",
+    "@type": "OrderQuote",
+    "brokerRole": "https://openactive.io/AgentBroker",
+    "broker": broker,
+    "seller": seller["@id"],
+    "orderedItem": [
+      {
+        "@type": "OrderItem",
+        "position": 0,
+        "acceptedOffer": offer["@id"],
+        "orderedItem": opportunity["@id"]
+      }
+    ]
+  };
+}
+
+function checkpointTwo(opportunity, seller, offer) {
+  return {
+    "@context": "https://openactive.io/",
+    "@type": "OrderQuote",
+    "brokerRole": "https://openactive.io/AgentBroker",
+    "broker": broker,
+    "seller": seller["@id"],
+    "customer": customer,
+    "orderedItem": [
+      {
+        "@type": "OrderItem",
+        "position": 0,
+        "acceptedOffer": offer["@id"],
+        "orderedItem": opportunity["@id"]
+      }
+    ]
+  }
+}
+
+function book(opportunity, seller, offer) {
+  return {
+    "@context": "https://openactive.io/",
+    "@type": "Order",
+    "brokerRole": "https://openactive.io/AgentBroker",
+    "broker": broker,
+    "seller": seller["@id"],
+    "customer": customer,
+    "orderedItem": [
+      {
+        "@type": "OrderItem",
+        "position": 0,
+        "acceptedOffer": offer["@id"],
+        "orderedItem": opportunity["@id"]
+      }
+    ]
+  }
+}
+
+function proposal(opportunity, seller, offer) {
+  return {
+    "@context": "https://openactive.io/",
+    "@type": "OrderProposal",
+    "brokerRole": "https://openactive.io/AgentBroker",
+    "broker": broker,
+    "seller": seller["@id"],
+    "customer": customer,
+    "orderedItem": [
+      {
+        "@type": "OrderItem",
+        "position": 0,
+        "acceptedOffer": offer["@id"],
+        "orderedItem": opportunity["@id"]
+      }
+    ]
+  }
+}
+
+function proposalUpdate(opportunity, seller, offer) {
+  return {
+    "@context": "https://openactive.io/",
+    "@type": "OrderProposal",
+    "orderProposalStatus": "https://openactive.io/CustomerRejected",
+    "orderCustomerNote": "Sorry I've actually made other plans, hope you find someone!"
+  }
+}
+
+function orderCancellation(opportunity, seller, offer) {
+  return {
+    "@context": "https://openactive.io/",
+    "@type": "Order",
+    "orderedItem": [
+      {
+        "@type": "OrderItem",
+        "@id": "",
+        "orderItemStatus": "https://openactive.io/CustomerCancelled"
+      }
+    ]
+  }
+}
+
+function buildSampleRequests(opportunity) {
+  const parent = opportunity.superEvent || opportunity.facilityUse;
+  const seller = parent.organizer || parent.provider;
+  const offer = parent.offers[0]
```

Comment:
This should use: https://github.com/openactive/openactive-test-suite/blob/ad0294105bd1dbcaf407fdf87c1380d69ae70c57/packages/openactive-integration-tests/test/helpers/flow-stages/fetch-opportunities.js#L46-L55 rather than just the first `Offer`, as the first `Offer` may not conform with the criteria.

Copying this hint from previous discussions:
> This little bit of logic could either be (a) centralised in the test-interface-criteria package alongside the getRelevantOffers function that it calls, or (b) just duplicated into the broker microservice - for the purposes of the endpoint we’re talking about creating
The opportunity that’s passed in is the full opportunity that you get from the second of the calls we looked at in the test suite output file